classes/patreon_login.php

```diff
@@ -89,54 +225,51 @@ public static function createUserFromPatreon($user_response, $tokens, $redirect
 				wp_set_auth_cookie( $user->ID );
 				do_action( 'wp_login', $user->data->user_login, $user );
 
-
-
 				/* update user meta data with patreon data */
-				self::updateExistingUser($user->ID, $user_response, $tokens);
+				self::updateExistingUser($user->ID, $user_response, $tokens);	
+				wp_redirect( $redirect );
+				exit;	
 
 			} else {
-				/* wordpress account creation failed #HANDLE_ERROR */
+				/* wordpress account creation failed */
+				
+				$redirect = add_query_arg( 'patreon_message', 'patreon_could_not_create_wp_account', $redirect);
+				wp_redirect( $redirect );
+				exit;			
+				
 			}
 
-		} else {
-
-			$danger_user_list = Patreon_Login::getDangerUserList();
-
-			if($login_with_patreon) {
-
-				if($admins_editors_login_with_patreon == false && array_key_exists($user->user_login, $danger_user_list) ) {
-
-					/* dont log admin / editor in */
-					wp_redirect( wp_login_url().'?patreon-msg=login_with_wordpress', '301' );
-					exit;
-
-				} else {
-
-					/* log user into existing wordpress account with matching email address */
-					wp_set_current_user( $user->ID, $user->user_login );
-					wp_set_auth_cookie( $user->ID );
-					do_action( 'wp_login', $user->user_login, $user);
-				}
-
-			}
-
-			/* update user meta data with patreon data */
-			self::updateExistingUser($user->ID, $user_response, $tokens);
+		}
+		else {
+				/* We created this patreon user before. Update and log in.
+			
+				/* update user meta data with patreon data */
+				wp_set_current_user( $user->ID, $user->data->user_login );
+				wp_set_auth_cookie( $user->ID );
+				do_action( 'wp_login', $user->data->user_login, $user );
 
+				/* update user meta data with patreon data */
+				self::updateExistingUser($user->ID, $user_response, $tokens);	
+				wp_redirect( $redirect );
+				exit;				
+			
 		}
+		
+		// Below code is probably dormant. #REVISIT
```

Comment:
did we revisit this? What was the verdict?

Reply:
if it's good, remove this comment to avoid confusion in the future

Reply:
Yes seems dormant indeed. The last block of ifs above them catch all cases. Removed.

https://github.com/codebard/patreon-wordpress/commit/82694036e01853877f00278e346b065bde4dd393